modules/claims_api/lib/claims_api/v2/disability_compensation_pdf_mapper.rb

```diff
@@ -94,11 +95,56 @@ def zip
         @pdf_data[:data][:attributes][:identificationInformation][:mailingAddress].merge!(zip:)
       end
 
+      def disability_attributes
+        @pdf_data[:data][:attributes][:claimInformation] = {}
+        @pdf_data[:data][:attributes][:claimInformation].merge!(
+          { disabilities: [] }
+        )
+        disabilities = transform_disabilities
+        details = disabilities[:data][:attributes][:claimInformation][:disabilities].map(
+          &:deep_symbolize_keys
+        )
+
+        @pdf_data[:data][:attributes][:claimInformation][:disabilities] = details
+
+        conditions_related_to_exposure?
+
+        @pdf_data
+      end
+
+      def transform_disabilities
+        d2 = []
+        claim_disabilities = @auto_claim['disabilities'].map do |disability|
+          disability['disability'] = disability['name']
+          disability.delete('name')
+          disability.delete('classificationCode')
+          disability.delete('ratedDisabilityId')
+          disability.delete('diagnosticCode')
+          disability.delete('disabilityActionType')
+          sec_dis = disability['secondaryDisabilities'].map do |secondary_disability|
+            secondary_disability['disability'] = secondary_disability['name']
+            secondary_disability
+          end
+          d2 << sec_dis
+          disability.delete('secondaryDisabilities')
+          disability
+        end
+        claim_disabilities << d2
+        @pdf_data[:data][:attributes][:claimInformation][:disabilities] = claim_disabilities.flatten.compact
+
+        @pdf_data
+      end
+
       def conditions_related_to_exposure?
         # If any disability is included in the request with 'isRelatedToToxicExposure' set to true,
         # set exposureInformation.hasConditionsRelatedToToxicExposures to true.
-        # This will check 'YES' for box 15A.
-        # TODO: for disability mapping ticket
+        @pdf_data[:data][:attributes][:exposureInformation][:hasConditionsRelatedToToxicExposures] = nil
+        has_conditions = @pdf_data[:data][:attributes][:claimInformation][:disabilities].any? do |disabiity|
```

Comment:
any returns a boolean here... looks like it just returns the first one (if it exists) as the `disability`

```
3.0.0 :028 > b = { disabilities: [{isReleatedToToxicExposure: false }, { isRelatedToToxicExposure: false }]}
 => {:disabilities=>[{:isReleatedToToxicExposure=>false}, {:isRelatedToToxicExposure=>false}]}
3.0.0 :029 > b[:disabilities]&.any? do |a|
3.0.0 :030 >   pp a
3.0.0 :031 > end
{:isReleatedToToxicExposure=>false}
 => true
```

Reply:
```irb(main):001:0> b = { disabilities: [{isReleatedToToxicExposure: false }, { isRelatedToToxicExposure: false }]}
=> {:disabilities=>[{:isReleatedToToxicExposure=>false}, {:isRelatedToToxicExposure=>false}]}
irb(main):002:0* b[:disabilities]&.any? do |a|
irb(main):003:0* pp a
irb(main):004:-> end
{:isReleatedToToxicExposure=>false}
=> true                                          
irb(main):005:0* b[:disabilities]&.any? do |a|
irb(main):006:0* a == true
irb(main):007:-> end
=> false```
I believe that this is working as intended. https://apidock.com/ruby/Enumerable/any%3F.
Now we have another test to see if the has_conditons attribute will show as false if there aren't any exposure related disabilities.